### PR TITLE
chore: Upgrade SDK version using pointer arrays in 5 resources

### PR DESCRIPTION
**Pull request by AgustinBettati**
## Proposed changes

Jira ticket: part of CLOUDP-222755

Updates SDK version (supporting pointer arrays) for the following resources:

- alert configuration (also includes setting up contract tests to run in CI)
- api key
- auditing
- cluster outage simulation
- online archive

**Note:** Given that changes are trivial do not see the need of doing a new release for each resource.

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as
  expected)
- [ ] This change requires a documentation update
- [ ] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

## Manual QA performed:

- [ ] cfn invoke for each of CRUDL/cfn test
- [ ] Updated resource in  [example](https://github.com/mongodb/mongodbatlas-cloudformation-resources/tree/master/examples)
- [ ] Published to AWS private registry
- [ ] Used the template in [example](https://github.com/mongodb/mongodbatlas-cloudformation-resources/tree/master/examples) to create and update a stack in AWS
- [ ] Deleted stack to ensure resources are deleted
- [ ] Created multiple resources in same stack
- [ ] Validated in Atlas UI
- [ ] Included screenshots

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [ ] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code
- [ ] For CFN Resources: I have released by changes in the private registry and proved by change
  works in Atlas

### CODE REVIEW

File: cfn-resources/alert-configuration/cmd/resource/resource.go [hunk 1]
```diff
@@ -300,20 +300,20 @@ func expandAlertConfigurationMetricThresholdConfig(currentModel *Model) *atlasSD
...
 
-func expandAlertConfigurationThreshold(threshold *IntegerThresholdView) *atlasSDK.GreaterThanRawThreshold {
+func expandAlertConfigurationThreshold(threshold *IntegerThresholdView) *admin.GreaterThanRawThreshold {
 	if threshold == nil {
 		return nil
 	}
-	return &atlasSDK.GreaterThanRawThreshold{
+	return &admin.GreaterThanRawThreshold{
 		Operator:  threshold.Operator,
 		Threshold: util.Pointer(int(*threshold.Threshold)),
```
Comment (lantoli) [inline cfn-resources/alert-configuration/cmd/resource/resource.go:309]: this seems a bit overcomplicated

File: cfn-resources/api-key/cmd/resource/resource.go [hunk 2]
```diff
@@ -461,23 +461,24 @@ func areStringArraysEqualIgnoreOrder(arr1, arr2 []string) bool {
...
+		if apiKeyRoles[i].OrgId != nil && apiKeyRoles[i].RoleName != nil {
+			roles = append(roles, *apiKeyRoles[i].RoleName)
 		}
 		// project roles
-		if apikey.Roles[i].GroupId != nil {
-			if apikey.Roles[i].RoleName != nil {
-				projectRolesMap[*apikey.Roles[i].GroupId] = append(projectRolesMap[*apikey.Roles[i].GroupId], *apikey.Roles[i].RoleName)
+		if apiKeyRoles[i].GroupId != nil {
+			if apiKeyRoles[i].RoleName != nil {
+				projectRolesMap[*apiKeyRoles[i].GroupId] = append(projectRolesMap[*apiKeyRoles[i].GroupId], *apiKeyRoles[i].RoleName)
```
Comment (lantoli) [inline cfn-resources/api-key/cmd/resource/resource.go:481]: can SDK Get functions like GetGroupId, GetRoleName, etc. be used?

**Review by lantoli — APPROVED** [review]
good!  is Atlas20231115002 still being used or it can be deleted?

File: cfn-resources/alert-configuration/cmd/resource/resource.go [hunk 3]
```diff
@@ -268,12 +268,12 @@ func List(req handler.Request, prevModel *Model, currentModel *Model) (handler.P
 }
 
 // function to check if record already exist
```
Comment (EspenAlbert) [inline cfn-resources/alert-configuration/cmd/resource/resource.go:270]: remove comment

**Review by EspenAlbert — APPROVED** [review]
LGTM!

File: cfn-resources/alert-configuration/cmd/resource/resource.go (hunk 1 above)
Comment (AgustinBettati) [reply cfn-resources/alert-configuration/cmd/resource/resource.go:309]: It does have the challenge of passing from float64 to int (CFN model defines as float64). This does raise a bigger point that we might be losing precision for this case, will take the action of seeing what type is in mms and if a fix is need in the SDK to be able to pass float values.

File: cfn-resources/alert-configuration/cmd/resource/resource.go (hunk 3 above)
Comment (lantoli) [reply cfn-resources/alert-configuration/cmd/resource/resource.go:270]: +1  :-)

File: cfn-resources/api-key/cmd/resource/resource.go (hunk 2 above)
Comment (AgustinBettati) [reply cfn-resources/api-key/cmd/resource/resource.go:481]: Prefer to leave this logic for handling the absence of specific properties in this case (vs using a potential empty string directly).

File: cfn-resources/alert-configuration/cmd/resource/resource.go (hunk 3 above)
Comment (AgustinBettati) [reply cfn-resources/alert-configuration/cmd/resource/resource.go]: removed quite a few :)

**Comment by AgustinBettati** [comment]
@lantoli 
> is Atlas20231115002 still being used or it can be deleted?

Still several resources to go!

**Pull request merged**